src/extra/others/controller/lv_obj_controller.c

```diff
@@ -0,0 +1,372 @@
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * item = item_new(cls);
+    lv_obj_t * parent = manager->container;
+    item_create_controller(manager, item, args);
+    /* Destroy object of previous screen */
+    if(manager->top) {
+        item_destroy_obj(manager, manager->top);
+    }
+    item_create_obj(manager, item, parent, NULL);
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+void lv_controller_manager_replace(lv_controller_manager_t * manager, const lv_obj_controller_class_t * cls,
+                                   void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * top = item_new(cls);
+    item_create_controller(manager, top, args);
+    manager_stack_t * old = manager->top;
+    if(old) {
+        item_destroy_obj(manager, old);
+        item_destroy_controller(old);
+        lv_mem_free(old);
+    }
+    manager->top = top;
+    item_create_obj(manager, top, manager->container, NULL);
+}
+
+size_t lv_controller_manager_size(lv_controller_manager_t * manager)
+{
+    size_t size = 0;
+    for(manager_stack_t * cur = manager->top; cur; cur = cur->prev) {
+        size++;
+    }
+    return size;
+}
+
+void lv_controller_manager_pop(lv_controller_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return;
+    manager_stack_t * prev = top->prev;
+    bool dialog = top->dialog;
+    if(!dialog && prev) {
+        item_create_controller(manager, prev, NULL);
+    }
+    item_destroy_obj(manager, top);
+    item_destroy_controller(top);
+    lv_mem_free(top);
+    if(!dialog && prev) {
+        item_create_obj(manager, prev, manager->container, NULL);
+    }
+    manager->top = prev;
+}
+
+bool lv_controller_manager_dispatch_event(lv_controller_manager_t * manager, int which, void * data1, void * data2)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return false;
+    lv_obj_controller_t * controller = top->controller;
+    if(!controller || !top->cls->event_cb) return false;
+    return top->cls->event_cb(controller, which, data1, data2);
+}
+
+#if LV_USE_MSGBOX
+
+void lv_controller_manager_show(lv_controller_manager_t * manager, const lv_obj_controller_class_t * cls, void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * item = item_new(cls);
+    item_create_controller(manager, item, args);
+    item_create_obj(manager, item, NULL, &lv_msgbox_class);
+    item->dialog = true;
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+#endif
+
+lv_obj_controller_t * lv_controller_manager_top_controller(lv_controller_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top)return NULL;
+    return top->controller;
+}
+
+lv_obj_controller_t * lv_controller_manager_parent(lv_controller_manager_t * manager)
```

Comment:
In the name of functions like this there should be a `get` word. E.g. `lv_controller_manager_get_parent`.

Shouldn't there be a `set` counterpart?

Reply:
Changed. This property is read only and specified upon creation so it can't have `set` counterpart.

Reply:
Thanks!